holyiot_yj16019 cannot compile IEEE 802.15.4 L2

**Describe the bug**
After applying #17477 the sanitychecker is complaining IEEE 802.15.4 compilation issues for holyiot_yj16019 board.

**To Reproduce**
* Apply #17477 
* `sanitycheck -p holyiot_yj16019 -s tests/net/ieee802154/l2/net.ieee802154.l2`

**Expected behavior**
Tests pass

**Additional context**
Some of the errors look like this, there are lot of them.
```
In file included from zephyr/drivers/ieee802154/ieee802154_nrf5.c:40:
zephyr/ext/hal/nordic/nrfx/mdk/nrf52840.h:71:3: error: redeclaration of enumerator 'Reset_IRQn'
   Reset_IRQn                = -15,              /*!< -15  Reset Vector, invoked on Power up and warm reset                     */
   ^~~~~~~~~~
In file included from zephyr/ext/hal/nordic/nrfx/mdk/nrf.h:88,
                 from zephyr/soc/arm/nordic_nrf/nrf52/soc.h:17,
                 from zephyr/include/arch/arm/cortex_m/cmsis.h:17,
                 from zephyr/include/arch/arm/cortex_m/mpu/arm_mpu_v7m.h:10,
                 from zephyr/include/arch/arm/cortex_m/mpu/arm_mpu.h:13,
                 from zephyr/include/arch/arm/arch.h:240,
                 from zephyr/include/arch/cpu.h:17,
                 from zephyr/include/kernel_includes.h:34,
                 from zephyr/include/kernel.h:17,
                 from zephyr/include/logging/log_msg.h:9,
                 from zephyr/include/logging/log_core.h:9,
                 from zephyr/include/logging/log.h:11,
                 from zephyr/drivers/ieee802154/ieee802154_nrf5.c:16:
zephyr/ext/hal/nordic/nrfx/mdk/nrf52.h:71:3: note: previous definition of 'Reset_IRQn' was here
   Reset_IRQn                = -15,              /*!< -15  Reset Vector, invoked on Power up and warm reset                     */
   ^~~~~~~~~~
```

--- a/boards/arm/holyiot_yj16019/doc/index.rst
+++ b/boards/arm/holyiot_yj16019/doc/index.rst
@@ -15,7 +15,7 @@ Semiconductor nRF52832 ARM Cortex-M4 CPU and the following devices:
 * :abbr:`MPU (Memory Protection Unit)`
 * :abbr:`NVIC (Nested Vectored Interrupt Controller)`
 * :abbr:`PWM (Pulse Width Modulation)`
-* RADIO (Bluetooth Low Energy and 802.15.4)
+* RADIO (Bluetooth Low Energy)
 * :abbr:`RTC (nRF RTC System Clock)`
 * Segger RTT (RTT Console)
 * :abbr:`WDT (Watchdog Timer)`
@@ -58,8 +58,7 @@ hardware features:
 +-----------+------------+----------------------+
 | PWM       | on-chip    | pwm                  |
 +-----------+------------+----------------------+
-| RADIO     | on-chip    | Bluetooth,           |
-|           |            | ieee802154           |
+| RADIO     | on-chip    | Bluetooth            |
 +-----------+------------+----------------------+
 | RTC       | on-chip    | system clock         |
 +-----------+------------+----------------------+
@@ -148,4 +147,3 @@ References
 
 .. _Holyiot: http://www.holyiot.com
 .. _Nordic Semiconductor Documentation library: https://www.nordicsemi.com/DocLib
-

--- a/boards/arm/holyiot_yj16019/holyiot_yj16019.yaml
+++ b/boards/arm/holyiot_yj16019/holyiot_yj16019.yaml
@@ -8,7 +8,6 @@ toolchain:
   - xtools
 supported:
   - ble
-  - ieee802154
   - pwm
   - watchdog
 ram: 64

